engine/resources/consul_kv.go

```diff
@@ -0,0 +1,186 @@
+// Mgmt
+// Copyright (C) 2013-2019+ James Shubin and the project contributors
+// Written by James Shubin <james@shubin.ca> and the project contributors
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package resources
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/purpleidea/mgmt/engine"
+	"github.com/purpleidea/mgmt/engine/traits"
+	"github.com/purpleidea/mgmt/util"
+)
+
+func init() {
+	engine.RegisterResource("consul:kv", func() engine.Res { return &ConsulKVRes{} })
+}
+
+// ConsulKVRes is a resource that writes a value into a consul datastore.
+type ConsulKVRes struct {
+	traits.Base
+	init *engine.Init
+
+	// Key is the path of the value.
+	Key string `lang:"key" yaml:"key"`
+
+	// Key is the value that will be written into consul.
+	Value string `lang:"value" yaml:"value"` // the consul key value
+
+	// Address is the address of the consul server.
```

Comment:
Can you add an example too, eg: "http://localhost:999" or whatever the common format is. Use example.com if you need a domain. Should this be named Server instead?